src/main/java/util/CarsParser.java

```diff
@@ -1,27 +1,28 @@
-package parser;
+package util;
 
 import entity.Car;
 import entity.Company;
+
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import java.util.ArrayList;
 import java.util.List;
-import java.util.concurrent.ArrayBlockingQueue;
-import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.Semaphore;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-import static parser.AttributeValueParser.COLOR;
-import static parser.AttributeValueParser.ID;
+import static util.AttributeValueParser.COLOR;
+import static util.AttributeValueParser.ID;
 
 public class CarsParser {
     private static final Logger LOGGER = LogManager.getLogger(Company.class);
     private static final AttributeValueParser parser = new AttributeValueParser();
     private static final String CAR_REGEX = "\\d+";
-    private static final int COMPANY_PARK_CAPACITY = 10;
+    private Semaphore semaphore = new Semaphore(1);
```

Comment:
Semaphore doesn't need to be stored in a Parser class. Just put **new Semaphore(1)** in **Car constructor**. But if you need only **1** value for the semaphore then have a look at the mutex.